src/main/java/us/myles/ViaVersion/ViaVersionPlugin.java

```diff
@@ -217,7 +237,10 @@ public synchronized void handleAdd(Object o) {
                         }
                     }
                 }
-            }
+
+            }.find(connection, connection.getClass(), ElementType.FIELD);
+
+            getLogger().info("Successfully injected!");
```

Comment:
@sinnlosername It's done in the method down below all this, essentially we inject ListWrappers which handle when the connection is added to the list later on :)